src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -758,6 +760,126 @@ private bool HasValidPlatformVersions()
+                    {
+                        bool doesPackageRemain = false;
+                        foreach (var framework in project.TargetFrameworks)
+                        {
+                            if (!prunedPackage.Value.Contains(framework.TargetAlias))
+                            {
+                                if (ContainsPackage(prunedPackage, framework))
+                                {
+                                    doesPackageRemain = true;
+                                    break;
+                                }
+                            }
+                        }
+                        if (doesPackageRemain)
+                        {
+                            continue;
+                        }
+                    }
+
+                    aliasToTargetGraphName ??= InitializeAliasToTargetGraphName(project);
+                    logger.Log(RestoreLogMessage.CreateWarning(
+                        NuGetLogCode.NU1510,
+                        string.Format(CultureInfo.CurrentCulture, Strings.Error_RestorePruningDirectPackageReference, prunedPackage.Key),
+                        prunedPackage.Key,
+                        prunedPackage.Value.Select(e => aliasToTargetGraphName[e]).ToArray()));
+                }
+            }
+
+            static bool HasFrameworkNewerThanNET10(PackageSpec project)
+            {
+                foreach (var framework in project.TargetFrameworks.NoAllocEnumerate())
+                {
+                    if (IsNewerThanNET10(framework.FrameworkName))
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            static Dictionary<string, string> InitializeAliasToTargetGraphName(PackageSpec project)
+            {
+                var aliasToTargetGraphName = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                foreach (var framework in project.TargetFrameworks)
+                {
+                    aliasToTargetGraphName.Add(framework.TargetAlias, FrameworkRuntimePair.GetTargetGraphName(framework.FrameworkName, runtimeIdentifier: null));
+                }
+
+                return aliasToTargetGraphName;
+            }
+
+            static bool ContainsPackage(KeyValuePair<string, List<string>> prunedPackage, TargetFrameworkInformation framework)
+            {
+                foreach (var dependency in framework.Dependencies.NoAllocEnumerate())
+                {
+                    if (dependency.Name.Equals(prunedPackage.Key, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            static bool IsNewerThanNET10(NuGetFramework frameworkName)
+            {
+                if (frameworkName.Framework == FrameworkConstants.FrameworkIdentifiers.NetCoreApp)
```

Comment:
If someone does `<TargetFramework>.netcoreapp,version=10.0</targetframework>`, does `frameworkName.Framework` use the original string, or the const string?  Because if it uses the original input string, this `==` is an ordinal compare, not a case insensitive compare.

Reply:
> does frameworkName.Framework use the original string, or the const string

It is the const string. 
I double checked with some of my aliasing tests and it was correct. 
We *always* translate to the pretty version, that's the whole point of the framework mappings.

That being said, I checked what IsNet5Era does, https://github.com/NuGet/NuGet.Client/blob/f61900e4d90c310e0c1615ab8db616cc65810859/src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs#L68. 

So I just decided to go ordinal ignore case as well. 
Better safe than sorry.